### DISCUSSION TITLE
Application using new API

### DISCUSSION THREAD

**Discussion by macbryla** [opennem/opennem#251] [General]
In light of the changes to the public API, I have a few questions regarding the data usage for my data visualisation project.

I put together a simple to understand digital display (see attached image) to show a LIVE view of the energy mix in the Australian electricity grid. The purpose for for this viz is to create an awareness of the level of renewables across the country. Currently, the application is standalone and not web-based. As a result, it's a single application making calls to your API at 5-minute intervals.

Based on the interest from the general public, I was planning to make a web-based version and make it freely available.

How would this work with the new API? 
Consider the following.
- The visualisation refreshes automatically every 5 minutes.
- It's web-based, and every instance will make calls to OpenNEM API. Not ideal.
- Currently, the calls are retrieving 7 days of data for every region at 5min or 30min resolution. This is a serious overkill, as all I need is only the latest values for each region and fuel type, i.e. a very tiny set of numbers. As for as I know, there is no API for this.
- Or a better option would be for me to make a single call, store the data and let clients query my database.

I am interested in your thoughts. 
Thanks.

<img width="1920" alt="AusPower" src="https://user-images.githubusercontent.com/251217/229428009-f165f7b3-727f-4fde-a99d-0ef721fc25c9.png">

**Comment by nc9** [comment]
The visualisation is awesome! 

>  I was planning to make a web-based version and make it freely available.

This would be great

> Currently, the calls are retrieving 7 days of data for every region at 5min or 30min resolution. This is a serious overkill, as all I need is only the latest values for each region and fuel type, i.e. a very tiny set of numbers. As for as I know, there is no API for this.

This is a good point - and should be supported. On the OpenNEM website it currently just refreshes the entire seven-day period as well. I'm planning on starting a wiki page with updates for the API

> Or a better option would be for me to make a single call, store the data and let clients query my database.

That sounds like too much overhead and your project is a perfect example of what we'd like to support directly, so we'll optimize for this use case

Have you [filled in the form](https://t.co/rHzJ0NVk71) so we can reach out to you with an API key?

**Reply by macbryla** [reply]
Hey Nik,

I have a test visualisation running as a single-page web app. Try it out. It automatically updates in your browser every 5min and currently uses the API described above, so it gets a lot of unnecessary data.

It's just a test until we find a more efficient way to get only the latest data needed.

It works well as a display to casually monitor the electricity mix.

Email me directly if you have any questions at mac@wiredlizard.com

https://macbryla.com/test/index.html

**Reply by nc9** [reply]
This is brilliant! Those `7d.json` files per-region are _definitely_ not appropriate for a use-case like this, which is something we want to support. I'll definitely figure out a better API interface to do 'current live' data out of the API.

Visualizations like yours are great at communicating the energy transition

**Reply by macbryla** [reply]
Thanks. That's the idea, to make it accessible to non-data people. 

Currently, is there an API to get data for the whole country (NEM + WA) ?

I can only see NEM
https://data.opennem.org.au/v3/stats/au/NEM/power/7d.json
and 
https://data.opennem.org.au/v3/stats/au/WEM/power/7d.json
and others for each state.

Thanks.

**Comment by nc9** [comment]
https://github.com/opennem/opennem/issues/252

**Comment by macbryla** [comment]
After a few iterations, here is a page with a link to the live version of the above data visualisation. I am still hoping for a more targeted API once v4 is out, which would not require to pull as much data with each request.

https://macbryla.com/data/aus-energy-transition/

**Reply by nc9** [reply]
This is brilliant! we're working on the API this week - shouldn't be long. you ready to make this public + promote it?
